baseband/mark5b/base.py

```diff
@@ -64,6 +92,40 @@ def read_frame(self):
                                     ref_time=self.ref_time, nchan=self.nchan,
                                     bps=self.bps)
 
+    def get_frame_rate(self):
+        """Returns the number of frames per second.
+
+        Returns
+        -------
+        frame_rate : `~astropy.units.Quantity`
+            Frames per second.
+
+        Notes
+        -----
+        This function first tries to determine the frame rate by looking for
+        the highest frame number in the first second of data.  If that fails,
+        it uses the time difference between two consecutive frames. This can
+        fail if the headers do not store fractional seconds, or if the data
+        rate is above 512 Mbps.
+        """
+        try:
+            return super(Mark5BFileReader, self).get_frame_rate()
+        except Exception as exc:
+            oldpos = self.tell()
+            self.seek(0)
+            try:
+                header0 = self.read_header()
+                self.seek(header0.payload_nbytes, 1)
+                header1 = self.read_header()
+                tdelta = header1.fraction - header0.fraction
+                assert tdelta != 0.
```

Comment:
Add a useful error message about this ("First two headers have identical timestamps, possibly because the file's data rate is above 512 Mbps.  Try explicitly passing in a sample rate.")?  I think it gets passed back to the stream reader.  Should also have a test.

(Obviously this is a strange edge case - when would we get a < 1 second file with that high a sampling rate?)